src/main/java/org/dependencytrack/tasks/scanners/SnykAnalysisTask.java

```diff
@@ -0,0 +1,467 @@
+        int trackComponent = 0;
+        final UnirestInstance ui = UnirestFactory.getUnirestInstance();
+        CountDownLatch countDownLatch = new CountDownLatch(components.size());
+        for (final Component component : components) {
+            if(trackComponent < components.size()){
+                trackComponent += 1;
+                Runnable analysisUtil = () -> {
+                    final String snykUrl = ApiBaseUrl + URLEncoder.encode(component.getPurlCoordinates().toString(), StandardCharsets.UTF_8) + API_ENDPOINT;
+                    if (!isCacheCurrent(Vulnerability.Source.SNYK, ApiBaseUrl, component.getPurl().toString())) {
+                        try {
+                            final GetRequest request = ui.get(snykUrl)
+                                    .header(HttpHeaders.AUTHORIZATION, this.apiToken);
+                            final HttpResponse<JsonNode> jsonResponse = request.asJson();
+                            if (jsonResponse.getStatus() == 200 || jsonResponse.getStatus() == 404) {
+                                handle(component, jsonResponse.getBody().getObject(), jsonResponse.getStatus(), jsonResponse.getStatusText());
+                            } else {
+                                handleUnexpectedHttpResponse(LOGGER, ApiBaseUrl, jsonResponse.getStatus(), jsonResponse.getStatusText());
+                            }
+                        } catch (UnirestException e) {
+                            handleRequestException(LOGGER, e);
+                        }
+                    } else {
+                        LOGGER.info("Cache is current, apply snyk analysis from cache");
+                        applyAnalysisFromCache(Vulnerability.Source.SNYK, ApiBaseUrl, component.getPurl().toString(), component, getAnalyzerIdentity(), vulnerabilityAnalysisLevel);
+                    }
+                    countDownLatch.countDown();
+                };
+                Instant startThread = Instant.now();
+                executor.execute(analysisUtil);
+
+                Instant endThread = Instant.now();
+                duration += Duration.between(startThread, endThread).toMillis();
+            }
+        }
+        try {
+            //System.out.println("countDownLatch" + countDownLatch.getCount());
+            countDownLatch.await();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+
+    public void handle(Component component, JSONObject object, int responseCode, String statusText) {
+
+        try (QueryManager qm = new QueryManager()) {
+            if (responseCode == 200) {
+                String purl = null;
+                final JSONObject metaInfo = object.optJSONObject("meta");
+                if (metaInfo != null) {
+                    purl = metaInfo.optJSONObject("package").optString("url");
+                    if (purl == null) {
+                        purl = parsePurlToSnykUrlParam(component.getPurl());
```

Comment:
This will URL encode the component's PURL, I don't think you want that here.

```suggestion
                        purl = component.getPurlCoordinates().toString();
```

Reply:
So we don't need method _parsePurlToSnykUrlParam_ at all?